src/Build.UnitTests/BackEnd/SdkResolverService_Tests.cs

```diff
@@ -217,6 +217,230 @@ public void CachingWrapperShouldWarnWhenMultipleVersionsAreReferenced()
+
+            var result = SdkResolverService.Instance.ResolveSdk(BuildEventContext.InvalidSubmissionId, sdk, _loggingContext, new MockElementLocation("file"), "sln", "projectPath", interactive: false);
+
+            result.Success.ShouldBeTrue();
+            result.Path.ShouldBe("Path/To/Return/From/Resolver");
+            result.Version.ShouldBe("1.0");
+
+            result.AdditionalPaths.ShouldBe(null);
+
+            ValidateExpectedPropertiesAndItems(true, result);
+
+            _logger.WarningCount.ShouldBe(0);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void SdkResultCanReturnMultiplePaths(bool includePropertiesAndItems)
+        {
+            string expectedPath1 = "First/Path/To/Return/From/Resolver";
+            string expectedPath2 = "Second/Path/To/Return/From/Resolver";
+
+            var sdk = new SdkReference("foo", "1.0", null);
+
+            Dictionary<string, string> propertiesToAdd = null;
+            Dictionary<string, SdkResultItem> itemsToAdd = null;
+
+            if (includePropertiesAndItems)
+            {
+                CreateMockSdkResultPropertiesAndItems(out propertiesToAdd, out itemsToAdd);
+            }
+
+            var resolver = new SdkUtilities.ConfigurableMockSdkResolver(
+                new SdkResultImpl(
+                    sdk,
+                    new []
+                    {
+                        new SdkResultPathAndVersion(expectedPath1, "1.0"),
+                        new SdkResultPathAndVersion(expectedPath2, "1.0")
+                    },
+                    propertiesToAdd,
+                    itemsToAdd,
+                    warnings: null
+                    ));
+
+            SdkResolverService.Instance.InitializeForTests(null, new List<SdkResolver>() { resolver });
+
+            var result = SdkResolverService.Instance.ResolveSdk(BuildEventContext.InvalidSubmissionId, sdk, _loggingContext, new MockElementLocation("file"), "sln", "projectPath", interactive: false);
+
+            result.Success.ShouldBeTrue();
+
+            var resultPaths = new List<SdkResultPathAndVersion>();
+            resultPaths.Add(new SdkResultPathAndVersion(result.Path, result.Version));
+            resultPaths.AddRange(result.AdditionalPaths);
+
+            resultPaths.ShouldBeEquivalentTo(new[]
+            {
+                new SdkResultPathAndVersion(expectedPath1, "1.0"),
+                new SdkResultPathAndVersion(expectedPath2, "1.0")
+            });
+
+            ValidateExpectedPropertiesAndItems(includePropertiesAndItems, result);
+
+            _logger.WarningCount.ShouldBe(0);
+        }
+
+        [Theory]
+        [InlineData(true, false)]
+        [InlineData(false, true)]
+        [InlineData(true, true)]
+        public void AssertResolutionWarnsIfResolvedVersionIsDifferentFromReferencedVersionWithMultipleReturnPaths(bool mismatch1, bool mismatch2)
```

Comment:
This tests that there's a warning generated when there's a mismatch between the requested SDK version and the one that's returned.  Now that each returned path doesn't have a separate version, the test is simplified - it's just a single `Fact` instead of a `Theory`.